client/app/account/components/DeleteAccount.tsx

```diff
@@ -1,45 +1,61 @@
+import React, { useState } from 'react'
 import { FiTrash } from 'react-icons/fi'
-import {
-  AlertDialog,
-  AlertDialogAction,
-  AlertDialogCancel,
-  AlertDialogContent,
-  AlertDialogDescription,
-  AlertDialogFooter,
-  AlertDialogHeader,
-  AlertDialogTitle,
-  AlertDialogTrigger,
-  Button,
-} from '@/_components/ui'
+import { Button } from '@/_components/ui'
+import { HouseType, UserType } from '@/_types'
 import { useDeleteAccount } from '../hooks'
+import { DeleteAccountModal, AdminWarningModal } from '.'
 
-const DeleteAccount = () => {
+type DeleteAccountProps = {
+  user: UserType
+  userHouse: HouseType | undefined
+}
+
+const DeleteAccount = ({ user, userHouse }: DeleteAccountProps) => {
+  const [isOpenDeleteModal, setIsOpenDeleteModal] = useState(false)
+  const [isOpenAdminWarningModal, setIsOpenAdminWarningModal] = useState(false)
   const { onDelete, isPending } = useDeleteAccount()
 
+  const handleDeleteAccount = () => {
+    if (!userHouse) {
+      setIsOpenDeleteModal(true)
+      return
+    }
+
+    const isAdmin = userHouse.houseMembers.filter((member) => member.isAdmin)
+
+    if (isAdmin.length === 1 && isAdmin[0].id === user.id) {
+      setIsOpenAdminWarningModal(true)
+    } else {
+      setIsOpenDeleteModal(true)
+    }
+  }
+
   return (
-    <AlertDialog>
-      <AlertDialogTrigger asChild>
-        <Button variant={'destructiveOutline'} size='sm' className='flex gap-1'>
-          <FiTrash />
-          Delete Account
-        </Button>
-      </AlertDialogTrigger>
-      <AlertDialogContent>
-        <AlertDialogHeader>
-          <AlertDialogTitle>Are you absolutely sure?</AlertDialogTitle>
-          <AlertDialogDescription>
-            This action cannot be undone. This will permanently delete your
-            account and remove the data from our servers.
-          </AlertDialogDescription>
-        </AlertDialogHeader>
-        <AlertDialogFooter>
-          <AlertDialogCancel disabled={isPending}>Cancel</AlertDialogCancel>
-          <AlertDialogAction isLoading={isPending} onClick={() => onDelete()}>
-            Delete
-          </AlertDialogAction>
-        </AlertDialogFooter>
-      </AlertDialogContent>
-    </AlertDialog>
+    <>
+      <Button
+        variant={'destructiveOutline'}
+        onClick={handleDeleteAccount}
+        size='sm'
+        className='flex gap-1'
+      >
+        <FiTrash />
+        Delete Your Account
+      </Button>
+      {isOpenDeleteModal && (
+        <DeleteAccountModal
+          isOpenDeleteModal={isOpenDeleteModal}
+          setIsOpenDeleteModal={setIsOpenDeleteModal}
+          onDelete={onDelete}
+          isPending={isPending}
+        />
+      )}
+      {isOpenAdminWarningModal && (
+        <AdminWarningModal
+          isOpenAdminWarningModal={isOpenAdminWarningModal}
+          setIsOpenAdminWarningModal={setIsOpenAdminWarningModal}
+        />
+      )}
```

Comment:
One question! Do you need `isOpenDeleteModal &&` and `isOpenAdminWarningModal &&`? You already pass the boolean to component so I believe you don't need them

Reply:
Yeah, your way is a duplicated condition, so you don't need to have this condition in this parent component. Child components have the condition already!

Reply:
Aha! Right!
b69dcc9a11a7a3645823e2b230a41d9cd75c6e69